arwen/interface.go

```diff
@@ -0,0 +1,62 @@
+package arwen
+
+import (
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"math/big"
+)
+
+type EthContext interface {
+	GetSCAddress() []byte
+	CallData() []byte
+	UseGas(gas int64)
+	GasLeft() int64
+	BlockGasLimit() int64
+	GetBalance(addr []byte) []byte
+	BlockHash(nonce int64) []byte
+	GetVMInput() vmcommon.VMInput
+	Transfer(destination []byte, sender []byte, value *big.Int, input []byte, gas int64) (gasLeft int64, err error)
+	GetStorage(addr []byte, key []byte) []byte
+	SetStorage(addr []byte, key []byte, value []byte) int32
+	GetCode(addr []byte) []byte
+	GetCodeSize(addr []byte) int32
+	SignalUserError()
+	Finish(data []byte)
+	WriteLog(addr []byte, topics [][]byte, data []byte)
+	SelfDestruct(addr []byte, beneficiary []byte)
+}
+
+type HostContext interface {
+	Function() string
+	Arguments() []*big.Int
+	GetStorage(addr []byte, key []byte) []byte
+	SetStorage(addr []byte, key []byte, value []byte) int32
+	GetBalance(addr []byte) []byte
+	BlockHash(nonce int64) []byte
+	GetVMInput() vmcommon.VMInput
+	GetSCAddress() []byte
+	WriteLog(addr []byte, topics [][]byte, data []byte)
+	Transfer(destination []byte, sender []byte, value *big.Int, input []byte, gas int64) (gasLeft int64, err error)
+	Finish(data []byte)
+	BlockChainHook() vmcommon.BlockchainHook
+	SignalUserError()
```

Comment:
For readability, sort these lines alphabetically, to highlight common methods with the `EthContext` interface.